### PR TITLE
Add Strip Sockets search action

### LINKED ISSUE #8419: "Strip sockets" search action
We've gotten some requests for an ability to bulk-reset items by removing their mods. I was thinking we could have a single action that reset all pluggable sockets to their initial state - mods, ornaments, shaders, etc. Thoughts?

### PR DESCRIPTION
**Pull request by robojumper**
Fixes #8419.

grafik

### CODE REVIEW

File: src/app/inventory/StripSockets.tsx [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+              onClick={() => {
+                if (activeKinds.includes(entry.kind)) {
+                  setActiveKinds(activeKinds.filter((k) => k !== entry.kind));
+                } else {
+                  setActiveKinds([...activeKinds, entry.kind]);
+                }
+              }}
+            />
+          ))
+        : 'No sockets to clear'}
```
Comment (delphiactual) [inline src/app/inventory/StripSockets.tsx:276]: i18n

**Comment by bhollis** [comment]
Overall I'm super into it. The sheet in place of a dialog is fine (I swear I'll make a dialog someday). I don't fully get why you'd reset subclasses though - I guess removing aspects and stuff?

**Comment by robojumper** [comment]
> I don't fully get why you'd reset subclasses though - I guess removing aspects and stuff?

Yeah I don't really see a use case either. That was just a strawman option I put in there as something this feature could theoretically do.

**Comment by robojumper** [comment]
After a short trip to stale closure hell (you can tell because I switched to a reducer), I think this is in a functional state now. Progress and error reporting uses a similar layout as the loadout application notification, but users can hover over failed mods or tap them to see the error.

**Review by bhollis — APPROVED** [review]

**Pull request merged**